### ISSUE TITLE
Does sample weights or scale_weight_pos affect leaf_value?

### ISSUE THREAD

**Issue by Bill671** [dmlc/xgboost#10001]
* I know that sample weight affects the loss function, which in turn affects the structure of the tree, but I haven't seen any studies on whether or not it affects the leaf values, and I have seen some studies that suggest that oversampling has the same effect as adjusting the weights.
* But if the calculation of the **leaf values** is based on **averaging**, then even if the model structure is the same, the calculated predictions should still be different, or is it the case that the leaf values are in fact based on **weighted averages**?
* Here is the simple example i would like to ask, suppose we have two sets of data x1,x2, the corresponding weights are w1,w2, and the corresponding residuals from the first round of training are $`\varepsilon _{1}^{2} ,\varepsilon _{2}^{2}`$ . (ps : the target is binary). Assuming that they are grouped into the same leaf node of the second decision tree, should the leaf value which included x1,x2 in the second decision tree  be $`(\varepsilon _{1}^{2} +\varepsilon _{2}^{2})/2`$, or $`(\varepsilon _{1}^{2}*w1+ \varepsilon _{2}^{2}*w2)`$?  Thks

**Comment by trivialfis** [comment]
I'm a bit confused by the notations used here. But the leaf value is calculated as $\sum{G} / (\sum{H} + \Omega)$, it's summation of gradient divided by the summation of hessian and regularization parameters. If the gradient is changed, the leaf value is also changed.

**Comment by Bill671** [comment]
> I'm a bit confused by the notations used here. But the leaf value is calculated as ∑G/(∑H+Ω), it's summation of gradient divided by the summation of hessian and regularization parameters. If the gradient is changed, the leaf value is also changed.

Thks,  you've given me some direction in finding the answer to that question. (ps : but i thinks it should be $'-\sum{G} / (\sum{H} + \Omega) '$ )

**Comment by trivialfis** [comment]
Excellent, feel free to reopen if there are further questions.
